src/ftpclass.cc

```diff
@@ -829,6 +827,45 @@ Ftp::pasv_state_t Ftp::Handle_EPSV()
    return PASV_HAVE_ADDRESS;
 }
 
+Ftp::pasv_state_t Ftp::Handle_EPSV_CEPR()
+{
+   char *format=alloca_strdup("(|%u|%[^'|']|%u|)");
```

Comment:
don't use alloca_strdup for format, it leaks memory here. Just put the format as an argument to sscanf.

Reply:
I directly pass it to sscanf now - I followed the pattern for EPSV (which usses alloca_strdup as well). Maybe that should be refacted then too?

Reply:
parsing EPSV have support for different separators, so the format is patched for that.